resolversrv/tests/unittests/test_db_query.py

```diff
@@ -0,0 +1,304 @@
+    # test
+    db_con = None
+
+    # Map of database connection parameters passed to the functions we're testing
+    db_conf = None
+
+    current_app = None
+
+    query_link_type = "SELECT * FROM public.datalinks WHERE bibcode = '{bibcode}' AND (link_type = '{link_type}')"
+    query_with_link_sub_type = "SELECT * FROM public.datalinks WHERE bibcode = '{bibcode}' AND (link_type = '{link_type}' AND link_sub_type = '{link_sub_type}')"
+
+    def create_app(self):
+        self.current_app = app.create_app()
+        return self.current_app
+
+    def setUp(self):
+        """ Module level set-up called once before any tests in this file are
+        executed.  Creates a temporary database and sets it up """
+        self.db = testing.postgresql.Postgresql()
+        # Get a map of connection parameters for the database which can be passed
+        # to the functions being tested so that they connect to the correct
+        # database
+        self.db_conf = self.db.dsn()
+        # Create a connection which can be used by our test functions to set and
+        # query the state of the database
+        self.db_con = psycopg2.connect(**self.db_conf)
+        # Commit changes immediately to the database
+        self.db_con.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        with self.db_con.cursor() as cur:
+            # Create the initial database structure (roles, schemas, tables etc.)
+            # basically anything that doesn't change
+            cur.execute(self.slurp('resolversrv/tests/data/datalinks.sql'))
+
+    def tearDown(self):
+        """ Called after all of the tests in this file have been executed to close
+        the database connecton and destroy the temporary database """
+        self.db_con.close()
+        self.db.stop()
+
+
+    def slurp(self, path):
+        """ Reads and returns the entire contents of a file """
+        with open(path, 'r') as f:
+            return f.read()
+
+    def convert(self, element):
+        results = []
+        for parts in element.lstrip('{').rstrip('}').replace('"', '').split(','):
+            results.append(parts)
+        return results
+
+    def execute_query(self, query):
+        with self.db_con.cursor() as cur:
+            cur.execute(query)
+            rows = cur.fetchall()
+            results = []
+            for row in rows:
+                bibcode = row[0]
+                link_type = row[1]
+                link_sub_type = row[2]
+                url = self.convert(row[3])
+                title = self.convert(row[4])
+                item_count = int(row[5])
+                results.append(DataLinks(bibcode, link_type, link_sub_type, url, title, item_count).toJSON())
+            return results
+        return ""
+
+
+    # returning a link
+    def test_link_presentation(self):
+        results = self.execute_query(self.query_link_type.format(bibcode='2017MNRAS.467.3556B', link_type='PRESENTATION'))
```

Comment:
app should have a `get_record` and `update_record` methods; you can use them to initialize the db (in the setUp()) and here test various responses

Reply:
done.